### ISSUE TITLE
Bash arithmetic expression which can fail

### ISSUE THREAD

**Issue by realsimix** [koalaman/shellcheck#2235]
#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

(( ${LANG+1} ))
(( ! ${LANG+1} ))
(( ${XXXX+1} ))
(( ! ${XXXX+1} ))

```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
I'm often using above expression to see if a variable is declared at all or not. In the example the first three expressions work but the last one doesn't `-bash: ((: !  : syntax error: operand expected (error token is " ")`.

It must be changed to `! (( ${XXXX+1} ))` to make it work.

It would be nice if such a case could be found by shellcheck.

Thanks for the very useful tool!
Simon
